pkg/controller/compute/managed.go

```diff
@@ -143,14 +146,42 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errNotAKSCluster)
 	}
 	cr.SetConditions(xpv1.Creating())
-	secret, err := e.newPasswordFn()
+
+	pw, err := e.getPassword(ctx, cr)
```

Comment:
We need to ask user for sensitive information through [an input `Secret`](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#sensitive-input-fields).

Reply:
Hi @muvaf,
I don't think it makes sense to make this an input parameter for the `AKSCluster`, though for an `Application` MR things would be different. Here, it's more of an output parameter as we provision an Application as a side-effect to support the AKSCluster.

One discussion around this was to make the Application itself part of the spec and not provision an Application if one is already provided. I have asked for a feature request for this as again, using an existing Application for the cluster is a new scenario that's currently not supported, and not sure if this is really something needed in the context of AKSClusters.

Reply:
I think the best way to deal with this is to split them apart to different CRDs and reference one from the other but that shouldn't block this PR.